### PR TITLE
fix NoSuchElementException when creating index

### PR DESCRIPTION
**Pull request by tarangbhalodia**
Fix for: https://github.com/sksamuel/elastic4s/issues/1992

### CODE REVIEW

**Comment by tarangbhalodia** [comment]
@sksamuel This PR is ready. Please let me know if any change required.

File: elastic4s-core/src/main/scala/com/sksamuel/elastic4s/requests/indexes/IndexTemplateHandlers.scala [hunk 1]
```diff
@@ -82,7 +82,7 @@ object CreateIndexTemplateBodyFn {
       create.settings.foreach {
         case (key, value) => builder.autofield(key, value)
       }
-      builder.rawField("analysis", AnalysisBuilder.build(create.analysis.get))
+      if(create.analysis.nonEmpty) builder.rawField("analysis", AnalysisBuilder.build(create.analysis.get))
```
Comment (Philippus) [inline elastic4s-core/src/main/scala/com/sksamuel/elastic4s/requests/indexes/IndexTemplateHandlers.scala]: how about avoiding the whole `.get` with:
```
      create.analysis.foreach { analysis =>
          builder.rawField("analysis", AnalysisBuilder.build(analysis))
      }
```

Comment (sksamuel) [reply elastic4s-core/src/main/scala/com/sksamuel/elastic4s/requests/indexes/IndexTemplateHandlers.scala]: Yep that would be better.

Comment (tarangbhalodia) [reply elastic4s-core/src/main/scala/com/sksamuel/elastic4s/requests/indexes/IndexTemplateHandlers.scala]: `.foreach` sounds good. Let me change it.

Comment (tarangbhalodia) [reply elastic4s-core/src/main/scala/com/sksamuel/elastic4s/requests/indexes/IndexTemplateHandlers.scala]: I've updated it.

**Comment by sksamuel** [comment]
:+1:

**Pull request merged**